python/revng/scripting/projects/project.py

```diff
@@ -0,0 +1,233 @@
+    ):
+        """
+        Run a single analysis. In addition to the `analysis_name` you need to specify a dict
+        of targets, some analysis require you to also pass an `options` dict.
+        """
+        pass
+
+    @abstractmethod
+    def analyses_list(self, analysis_name: str):
+        """
+        Run analysis list, these are predefined list of analysis that run sequentially.
+        """
+        pass
+
+    @abstractmethod
+    def commit(self):
+        """
+        Persist the changes from `self.model` to the backend.
+        """
+        pass
+
+    def revert(self):
+        """
+        Revert changes made to the `self.model` since the last call to `commit()`.
+        """
+        self._set_model(deepcopy(self.last_saved_model))
+
+    @abstractmethod
+    def _get_pipeline_description(self):
+        """
+        Define how to implement getting the pipeline description data, The method should parse the
+        data as `yaml` and store it in the variable `self.pipeline_description`
+        """
+        pass
+
+    def import_and_analyze(self, input_binary_path: str):
+        """
+        A helper method for setting up the project. This method imports
+        the binary and runs the `revng` initial auto analysis.
+        """
+        self.import_binary(input_binary_path)
+        self.analyses_list("revng-initial-auto-analysis")
+        self.analyses_list("revng-c-initial-auto-analysis")
+        self._get_pipeline_description()
+        self._set_model_mixins()
+
+    def get_analysis_inputs(self, analysis_name: str) -> Dict[str, List[str]]:
+        """
+        Get the analysis container inputs and the associate acceptable kinds from
+        the `pipeline description`.
+        """
+        inputs: Dict[str, List[str]] = {}
+        for step in self.pipeline_description["Steps"]:
+            for analysis in step["Analyses"]:
+                if analysis["Name"] == analysis_name:
+                    for i in analysis["ContainerInputs"]:
+                        inputs[i["Name"]] = i["AcceptableKinds"]
+        return inputs
+
+    def _set_revng_executable(self, revng_executable_path: str | None) -> str:
+        """
+        Check if the path of the user supplied or default revng executable
+        is present on the system
+        """
+        if not revng_executable_path:
+            revng_executable = shutil.which("revng")
+        else:
+            revng_executable = revng_executable_path
+        assert revng_executable is not None
+        assert os.access(revng_executable, os.F_OK | os.X_OK)
+        return revng_executable
+
+    def _set_resume_dir(self, resume_dir: str | None) -> str:
+        """
+        Use the user supplied resume dir or create a temp dir
+        """
+        if not resume_dir:
+            self._tmp_dir = TemporaryDirectory()
+            resume_dir = self._tmp_dir.name
+        assert os.path.isdir(resume_dir)
+        return resume_dir
```

Comment:
The function name and the semantics are a bit weird, I'd make these two functions actually set the the variable in `self` and not return anything

Reply:
Yeah the name remained after I changed the functionality to return the values, I will make the functions est the values instead.